reference/src/layout/structs-and-tuples.md

```diff
@@ -79,6 +79,11 @@ themselves is already entirely determined by their types, and since we intend to
 allow creating references to fields (`&s.f1`), structs do not have any
 wiggle-room there.
 
+**Note:** This is only true if the struct is inhabited.  For structs like `(i32,
+!)` that do not have a valid inhabitant, the compiler has more freedom.  After
+all, no references to fields can ever be taken.  For example, such structs might
+be zero-sized.
+
```

Comment:
This makes sense and I suppose we could actually make this a guarantee already, but I'm left wondering if this guarantee "propagates" back to `!`. E.g. if `struct A(!)` can be zero-sized, does that mean that `!` can be zero-sized ?

Reply:
`!` is zero-sized... so yes it can be...

Reply:
Could we word that as well somewhere? Maybe add a `uninhabited.md` document saying at least that ?

Reply:
Ah you reworded things, LGTM.